frontend/packages/dev-console/src/components/edit-deployment/deployment-strategy/DeploymentStrategySection.tsx

```diff
@@ -21,30 +23,59 @@ const DeploymentStrategySection: React.FC<DeploymentStrategySectionProps> = ({
   const { t } = useTranslation();
   const {
     values: {
-      deploymentStrategy: { type },
+      formData: {
+        name: resName,
+        project: { name: resNamespace },
+        deploymentStrategy: { type },
+      },
     },
+    initialValues,
+    setFieldValue,
   } = useFormikContext<FormikValues>();
 
   const deploymentStrategyFields = React.useMemo(() => {
     switch (type) {
-      case DeploymentStrategyType.recreate:
-        return <RecreateStrategy resourceType={resourceType} />;
-      case DeploymentStrategyType.rolling:
-        return <RollingStrategy resourceType={resourceType} />;
-      case DeploymentStrategyType.custom:
+      case DeploymentStrategyType.recreateParams:
+        return <RecreateStrategy resourceType={resourceType} resourceObj={resourceObj} />;
+      case DeploymentStrategyType.rollingParams:
+      case DeploymentStrategyType.rollingUpdate:
+        return <RollingStrategy resourceType={resourceType} resourceObj={resourceObj} />;
+      case DeploymentStrategyType.customParams:
         return <CustomStrategy resourceObj={resourceObj} />;
       default:
         return null;
     }
-  }, [resourceObj, resourceType, type]);
+  }, [type, resourceObj, resourceType]);
 
   return (
     <FormSection title={t('devconsole~Deployment strategy')}>
       <DropdownField
-        name="deploymentStrategy.type"
+        name="formData.deploymentStrategy.type"
         label={t('devconsole~Strategy type')}
         items={DeploymentStrategyDropdownData[resourceType].items}
         helpText={DeploymentStrategyDropdownData[resourceType].helpText[type]}
+        onChange={(value) => {
+          const strategyDefaultValues = getStrategyData(
+            value,
+            {},
+            resName,
+            resNamespace,
+            resourceType,
+          );
+          const strategyData = {
+            ...(_.omit(resourceObj.spec?.strategy, [
+              'rollingParams',
+              'recreateParams',
+              'customParams',
+              'rollingUpdate',
+            ]) ?? {}),
+            type: value,
+            ...strategyDefaultValues,
+          };
+          initialValues.formData.deploymentStrategy = strategyData;
```

Comment:
Not sure you ever want to mutate initialValues after the first usage. Formik does support updating to them but I'd reckon that's not something we should ever do.

Reply:
In this case its needed to mutate initialValues and we are doing it at other places as well.

Reply:
Hmmm... interesting. Not sure it brings value, if it's not ignored we likely are doing questionable stuff with our `initialValues` as they stop being 'initial' when we can mutate them and change things. What happens if you remove this line?

Reply:
When imageStream component is rendered it takes values from initialValues and populate the dropdowns accordingly. I am updating the initialValues to add image-stream values when the user adds the values in tagImages section of the lifecyclehook form so that when the user closes the form or navigates from yaml-editor to form-editor the previously selected values still show up.